src/containerapp/azext_containerapp/custom.py

```diff
@@ -2197,3 +2201,109 @@ def show_env_managed_identity(cmd, name, resource_group_name):
         r["identity"] = {}
         r["identity"]["type"] = "None"
         return r["identity"]
+
+
+def list_java_components(cmd, environment_name, resource_group_name):
+    raw_parameters = locals()
+    java_component_decorator = JavaComponentDecorator(
+        cmd=cmd,
+        client=JavaComponentPreviewClient,
+        raw_parameters=raw_parameters,
+        models=CONTAINER_APPS_SDK_MODELS
+    )
+    return java_component_decorator.list()
+
+
+def show_java_component(cmd, java_component_name, environment_name, resource_group_name, target_java_component_type):
+    raw_parameters = locals()
+    java_component_decorator = JavaComponentDecorator(
+        cmd=cmd,
+        client=JavaComponentPreviewClient,
+        raw_parameters=raw_parameters,
+        models=CONTAINER_APPS_SDK_MODELS
+    )
+    result = java_component_decorator.show()
+
+    current_type = safe_get(result, "properties", "componentType")
+    if current_type and target_java_component_type.lower() != current_type.lower():
+        raise ResourceNotFoundError(f"(JavaComponentNotFound) JavaComponent '{java_component_name}' was not found.")
+
+    return result
+
+
+def delete_java_component(cmd, java_component_name, environment_name, resource_group_name, target_java_component_type, no_wait):
+    raw_parameters = locals()
+    java_component_decorator = JavaComponentDecorator(
+        cmd=cmd,
+        client=JavaComponentPreviewClient,
+        raw_parameters=raw_parameters,
+        models=CONTAINER_APPS_SDK_MODELS
+    )
+
+    result = None
+    try:
+        result = java_component_decorator.client.show(cmd, resource_group_name, environment_name, java_component_name)
```

Comment:
Here I have discussed with shichao, because if use `java_component_decorator.show()`, the `status_code` will miss, so recommend using `java_component_decorator.client.show` directly.